### DISCUSSION TITLE
Tracking a mobile app

### DISCUSSION THREAD

**Discussion by jaxkodex** [umami-software/umami#2846] [Q&A]
Hello, I am considering using `umami` for tracking website visitor and the product seems very complete! I was wondering though if I could also use it to track analytics in a mobile app, technically I think I could just call the track API, however I am not sure if that's an expected approach, plus it is a bit weird there is no mobile sdk provided yet. Is tracking mobile apps something that can be done?

**Comment by everrr1025** [comment]
i have the same issue in my project, however as umami says, it focus on webside data,  i dont think there will have any sdk for mible app in the near future. i am trying to develop my own mobile sdk which has umami api integrated, and it will work technically.

**Comment by zeroCoder1** [comment]
I’ve been working on evaluating this for a mobile app, and I’ve found it to be quite straightforward to set up. 

When you’re tracking a mobile app, you’ll need to send the events in batches, just before the app is dismissed or suspended. 

There was a missing batch API for this, but I opened a PR and it’s now merged into the develop branch. 

On the one hand, it’s great that there’s no SDK involved, but it’s not a huge amount of work to set this up and run it for mobile apps.

I am figuring out to make an SDK for native apps, I would need some help though.

**Reply by jaxkodex** [reply]
I also considered this, however I went a different route in the end, and started using Jitsu for mobile. It already has SDKs and I'll manually analyze data later.

**Reply by zeroCoder1** [reply]
you can you mixpanel SDK, it has internal proxy that you use to post events to your server.

**Comment by jaxkodex** [comment]
Closing the discussion as it seems like it is not one of the goals of the project and other tools can be used to track mobile.

**Comment by MessiasLima** [comment]
Hey @jaxkodex How are you? I made this library that aims to integrate umami analytics with Kotlin application https://umami-kotlin.appoutlet.dev/ . Im not sure about the tech stack of your app, but I think my lib can help you
